crates/nargo_cli/src/preprocess.rs

```diff
@@ -0,0 +1,57 @@
+use acvm::ProofSystemCompiler;
+use iter_extended::vecmap;
+use noirc_driver::{CompiledContract, CompiledProgram};
+
+// TODO: migrate to `nargo_cli`
+
+use crate::artifacts::{
+    contract::{PreprocessedContract, PreprocessedContractFunction},
+    program::PreprocessedProgram,
+};
+
+// TODO: pull this from backend.
```

Comment:
Can this be done before merging?

Reply:
The backend doesn't currently expose this (and we can't push a new release without solving UP).

Reply:
Makes sense -- lets create a tracking issue for this

Reply:
https://github.com/noir-lang/acvm/issues/196